app/controllers/microsoft/callbacks_controller.rb

```diff
@@ -45,13 +45,20 @@ def find_or_create_inbox
     channel_email.inbox
   end
 
+  # Fallback name, for when name field is missing from users_data
+  def fallback_name
+    return 'New Inbox' if users_data['email'].blank?
+
```

Comment:
do we need this condition. users_data['email'] can't be blank in any case right. we will need it for creating the inbox.

Reply:
I think this is just extra check that they have added, but this will not and should not be the case.

Let me update it.